### ISSUE TITLE
Feature: Make it easier to work with S3 Express ONE Zone

### ISSUE THREAD

**Issue by resonating-sirsh** [lancedb/lancedb#1206]
### SDK

Python

### Description

Hello - ive been trying to use an S3 Express Directory bucket as a a DB root. 
By adding an environment variable that lance understands it seems its OK to create tables/datasets in this way
```python
import os
os.environ["aws_s3_express"] = "true"
```
But then when I try to re-open the lanecdb tables which I can see on S3, I get a file not found. It seems if I dig through the code it is related to how [pyarrow s3 file system](https://arrow.apache.org/docs/python/generated/pyarrow.fs.S3FileSystem.html) tries to list objects in these bucket types. I noticed in the lancedb code there was an `AWS_ENDPOINT` env var which can be passed in and used to override same in pyarrow file system constructor and that seemed promising to configure s3 express zonal endpoints but things break down when I try to get the pyarrow file system to work that way.
anyway, TL;DR i have no idea what im doing but perhaps there is a recipe for using S3 express that can be developed or made more clear in the docs?

**Comment by wjones127** [comment]
Hi, yeah the PyArrow filesystem stuff will mess with this right now. `AWS_ENDPOINT` is different from the S3 Express stuff, so I don't think there's a way to get that to work.

You will likely have better luck using the async API, which doesn't use any PyArrow filesystems. Use `connect_async` instead of `connect()`. We'll be moving over the sync API to wrap the async one later.

**Comment by resonating-sirsh** [comment]
> Hi, yeah the PyArrow filesystem stuff will mess with this right now. `AWS_ENDPOINT` is different from the S3 Express stuff, so I don't think there's a way to get that to work.
> 
> You will likely have better luck using the async API, which doesn't use any PyArrow filesystems. Use `connect_async` instead of `connect()`. We'll be moving over the sync API to wrap the async one later.

excellent - thanks for the advice.
